nixos/tests/installed-tests/default.nix

```diff
@@ -0,0 +1,68 @@
+# NixOS tests for gnome-desktop-testing-runner using software
+
+{ system ? builtins.currentSystem,
+  config ? {},
+  pkgs ? import ../../.. { inherit system config; }
+}:
+
+with import ../../lib/testing-python.nix { inherit system pkgs; };
+with pkgs.lib;
+
+let
+
+  callInstalledTest = pkgs.newScope { inherit makeInstalledTest; };
+
+  makeInstalledTest =
+    { tested               # Package to test. Needs to have an installedTests output
+    , testConfig ? {}      # Config to inject into machine
+    , extraTestScript ? "" # Test script snippet to inject before gnome-desktop-testing-runnner begins
+    , withX11 ? false      # Does test need X11?
+    , testRunnerFlags ? "" # Extra flags to pass to gnome-desktop-testing-runner
```

Comment:
Maybe we should make this an array in line with structuredAttrs proposal. I can imagine python test driver to accept list of args like Python’s `subprocess` module does.